File: Estructura_de_datos_C#/Dato_Abstracto.cs
using System;

public class Persona
{
    public string Nombre;
    public string Apellido;
    public int Edad;

    public Persona(string n, string a, int e)
    {
        Nombre = n;
        Apellido = a;
        Edad = e;
    }
}

class Program
{
    static void Main(string[] args)
    {
        Persona[] personas = new Persona[3]
        {
            new Persona("Jose", "Aguallo", 30),
            new Persona("Sheila", "Valdez", 25),
            new Persona("Luis", "Ramírez", 40)
        };

        for (int i = 0; i < personas.Length; i++)
        {
            Console.WriteLine($"Persona {i + 1}");
            Console.WriteLine("Nombre: " + personas[i].Nombre);
            Console.WriteLine("Apellido: " + personas[i].Apellido);
            Console.WriteLine("Edad: " + personas[i].Edad);
            Console.WriteLine("------------------");
        }
    }
}

File: Estructura_de_datos_Java/Dato_abstracto.java
public class Dato_abstracto {
    // Clase Persona (tipo de dato definido por el usuario)
    static class Persona {
        String nombre;
        String apellido;
        int edad;

        // Constructor
        public Persona(String n, String a, int e) {
            nombre = n;
            apellido = a;
            edad = e;
        }
    }

    public static void main(String[] args) {
        // Crear un arreglo de 3 personas
        Persona[] personas = new Persona[3];

        // Asignar valores a cada posición del arreglo
        personas[0] = new Persona("Jose", "Aguallo", 30);
        personas[1] = new Persona("Sheila", "Valdez", 25);
        personas[2] = new Persona("Luis", "Ramírez", 40);

        // Recorrer el arreglo y mostrar los datos
        for (int i = 0; i < personas.length; i++) {
            System.out.println("Persona " + (i+1));
            System.out.println("Nombre: " + personas[i].nombre);
            System.out.println("Apellido: " + personas[i].apellido);
            System.out.println("Edad: " + personas[i].edad);
            System.out.println("--------------------");
        }
    }
}


